app/flatpak-builtins-repo-update.c

```diff
@@ -317,7 +316,7 @@ generate_all_deltas (OstreeRepo *repo,
                               cancellable, error))
     return FALSE;
 
-  g_main_context_push_thread_default (context);
+  g_autoptr(GMainContextPopDefault) context = flatpak_main_context_new_default ();
```

Comment:
This is going to cause problems, because gcc is stupid with declarations like this. The variable is owned by the scope, but initializated at this point in the code. This means early returns will free an un-iniitialized context variable.

So, move variable to top and NULL init it there.

Reply:
Done.